modules/services/getmail.nix

```diff
@@ -0,0 +1,205 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.getmail;
+
+  retrieverModule = types.submodule ({config,...}: {
+    options = {
+      type = mkOption {
+        type = types.enum [
+          "SimplePOP3Retriever"
+          "SimplePOP3SSLRetriever"
+          "SimpleIMAPRetriever"
+          "SimpleIMAPSSLRetriever"
+        ];
+        default = "SimpleIMAPSSLRetriever";
+        description = "Type of the retriever.";
+      };
+
+      server = mkOption {
+        type = types.string;
+        default = "";
+        description = "The remote server.";
+      };
+
+      username = mkOption {
+        type = types.string;
+        default = "";
+        description = "The server username.";
+      };
+
+      password = mkOption {
+        type = types.nullOr types.string;
+        default = null;
+        description = ''
+          The server password. Note that the passwords are stored clear in the
+          nix store, so it is recommended to not use this field, but instead
+          either leave empty or use <literal>passwordCommand</literal> instead.
+        '';
+      };
+
+      passwordCommand = mkOption {
+        type = types.nullOr (types.listOf types.string);
+        default = null;
+        example = ["${pkgs.gnupg}/bin/gpg" "--decrypt" "file.gpg"];
```

Comment:
I think it might be better to have this of type `nullOr str` similar to a services `ExecStart`, then the user won't have to think about escaping stuff (e.g. would setting `passwordCommand = ["foo" "bar baz"]` invoke it with 2 or 3 arguments?)